src/js/providers/providers.js

```diff
@@ -62,6 +55,9 @@ Object.assign(Providers.prototype, {
         // prevent throw on missing source
         source = (source === Object(source)) ? source : {};
 
+        if (source.type === 'youtube' || source.type === 'rtmp') {
+            return null;
```

Comment:
@karimJWP This `choose` method's job is to see if any providers support the source and return that provider. It should not make any decisions like this itself. Providers can be added to the player that have their own logic.